### ISSUE TITLE
Add warnings for `sh` for non-POSIX extensions

### ISSUE THREAD

**Issue by shanebishop** [koalaman/shellcheck#2125]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

This is just a small example, please read **More details** section below.

```sh
#!/bin/sh 
[ -k foo ]
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
In POSIX sh, -k is undefined.

(Or something similar.)

#### More details

Shellcheck does not emit warnings for the non-POSIX operators `-nt`, `-ef`, `-k`, `-o`, `<`, and `>` when using a `#!/bin/sh` shebang or `--shell sh`.

Shellcheck already emits warnings for non-POSIX syntax ([example 1](https://github.com/koalaman/shellcheck/wiki/SC2039), [example 2](https://github.com/koalaman/shellcheck/wiki/SC3026)). It would be useful if shellcheck could also warn for these non-POSIX operators, when the `sh` shell is specified.
